Configure

```diff
@@ -132,15 +132,19 @@ my $gcc_devteam_warn = "-DDEBUG_UNUSED"
         . " -Werror"
```

Comment:
Do you really need $devteam_common_warn to contain -Werrror? To supress compiler warnings in icc, we need to remove -Werror. I think -Werror should be present only in gcc and clang.

Reply:
We prefer to add extra compiler specific options to disable select warnings.  Is that not possible with icc?

Reply:
Ok will check tomorrow and get back to you.

Reply:
Taking @richsalz's snippet in https://github.com/openssl/openssl/pull/4248#issuecomment-324909087, compiling with -Werror -Wswitch -Wno-switch works. So that as long as additional flags are added *after* default flags, it's always possible to override with -Wno-whatever. You can even override -Werror with later -Wno-error...

Reply:
I think @dot-asm has the right approach.  Adding `-Wno-swtich` seems better than removing `-Wall`.